npg_qc_viewer/t/client/test_uqc_outcomes_view.js

```diff
@@ -0,0 +1,217 @@
+        function() {
+          qc_outcomes.utilityDisplaySwitch();
+        },
+        /Parameter obj must be defined and of type \(DOM\) Element/i,
+        'Throws with no params'
+      );
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(1);
+        },
+        /Parameter obj must be defined and of type \(DOM\) Element/i,
+        'Throws with one (non DOM) param'
+      );
+      var some_element = $('td.lane')[0];
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(some_element);
+        },
+        /utility should be a string/i,
+        'Throws with one param'
+      );
+      assert.throws(
+        function() {
+          qc_outcomes.utilityDisplaySwitch(some_element, true);
+        },
+        /utility should be a string/i,
+        'Throws with non string parameter'
+      );
+    });
+
+    QUnit.test('Updating for a uqc element with undefined uqc outcome', function(assert) {
+      var qcOutcomes = {"lib":{},
+                        "seq":{},
+                        "uqc":{"19001:1":{}}};
+      assert.throws(
+        function() {
+          qc_outcomes._updateDisplayWithQCOutcomes(qcOutcomes);
+        },
+           /Malformed QC outcomes data for 19001:1/i,
+           'Throws with empty uqc_outcome'
+      );
+    }); 
+
+    QUnit.test('Updating for a uqc element with unrecognized uqc outcome', function(assert) {
+      var qcOutcomes = {"lib":{},
+                        "seq":{},
+                        "uqc":{"19001:1":{"uqc_outcome":"RandomOutcome"}}};
+      assert.throws(
+        function() {
+          qc_outcomes._updateDisplayWithQCOutcomes(qcOutcomes);
+        },
+           /Unknown utility/i,
+           'Throws with empty uqc_outcome'
+      );
+    }); 
+
+    QUnit.test('Updating for a non-existing rpt_key', function(assert) {
+      var qcOutcomes = {"lib":{},
+                        "seq":{"18245:1":{"mqc_outcome":"Rejected final"},
+                               "19001:1":{"mqc_outcome":"Accepted final"}},
+                        "uqc":{"18245:1":{"uqc_outcome":"Accepted"},
+                               "19001:1":{"uqc_outcome":"Rejected"},
+                               "1824500:1":{"uqc_outcome":"Accepted"}}};
+
+      var rpt_keys = Object.keys(qcOutcomes.uqc);
+      var received = {};
+      for (var i in rpt_keys) {
+        var key = rpt_keys[i];
+        var elementClass = qc_utils.isLaneKey(key) ? 'lane' : 'tag_info';
+        var arr = {};
+        arr[key] = qcOutcomes.uqc[key];
+        received[key] = qc_outcomes._processOutcomes( arr, 'uqc_outcome', elementClass)[key];
+      }
+      var expected = {"18245:1":1,"19001:1":1,"1824500:1":0};
+      assert.deepEqual(received, expected, 
+        '__processOutcomes reports expected hash with 2 existing and 1 absent element');
+    }); 
+
+    QUnit.test("utilityDisplaySwitch tests for FAIL/PASS/UNDECIDED utility icon in LaneNo column", function (assert) {
```

Comment:
Here and further: the tests are very elaborate. It would have been much easier and more explicit for each element under test to have an id and to be found by one when a change to the element has to be examined. The selectors in tests are practically the same as in the code under the test, which is unfortunate.